### PR TITLE
Folly enabled by default

### PR DESCRIPTION
**Pull request by kurapov-peter**
The failures we saw were cause by cmake picking up dependencies via the 'module path'. Also, Folly searches for its dependencies and overwrites `Boost_LIBRARIES` which caused linking issues in the logger and hence almost everywhere. I removed the find-modules altogether since they are outdated anyway.

The version of Folly is the same as we have in conda.

### CODE REVIEW

File: omniscidb/CMakeLists.txt [hunk 1]
```diff
@@ -309,19 +309,8 @@ find_package(ZLIB REQUIRED)
...
 if(ENABLE_FOLLY)
-  find_package(Folly)
-  if(NOT Folly_FOUND)
-    set(ENABLE_FOLLY OFF CACHE BOOL "Use Folly" FORCE)
-  else()
-    include_directories(${Folly_INCLUDE_DIRS})
-    add_definitions("-DHAVE_FOLLY")
-    list(APPEND Folly_LIBRARIES ${Glog_LIBRARIES})
-    # TODO: use Folly::folly_deps?
-    if(MSVC)
```
Comment (ienkovich) [inline omniscidb/CMakeLists.txt:320]: Are you sure this fix for Windows is not required anymore?

Comment (kurapov-peter) [reply omniscidb/CMakeLists.txt:320]: They link it in the dependencies https://github.com/facebook/folly/blob/main/CMake/folly-deps.cmake#L70
I also saw libevent in the list of linked libraries in `folly-targets.cmake` that's installed.
I'm not 100% sure, but should work.

Comment (ienkovich) [reply omniscidb/CMakeLists.txt:320]: Sounds good. Thanks!

**Comment by ienkovich** [comment]
Does folly enabling break modin?

**Comment by kurapov-peter** [comment]
> Does folly enabling break modin?

So it seems. The failure looks stable and happens in the `ArrowStorage::createTable`, and ArrowStorage does depend on folly, but I don't yet know what's the cause for a segfault. Guess I'll have to reproduce it.

**Comment by ienkovich** [comment]
> So it seems. The failure looks stable and happens in the `ArrowStorage::createTable`, and ArrowStorage does depend on folly, but I don't yet know what's the cause for a segfault. Guess I'll have to reproduce it.

You can try PyHDK tests which also use ArrowStorage from Python. Rebase on the latest main branch should add it to CI BTW.

**Comment by kurapov-peter** [comment]
Looks like we are corrupting the state of `ArrowStorage`. Here's the stack trace:
```
#0  0x00007fffa25f1ba4 in std::_Hashtable<int, std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > >, std::allocator<std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > >, std::__detail::_Select1st, std::equal_to<int>, std::hash<int>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_find_before_node (__code=0, __k=@0x55555709c378: 0, __n=0, 
    this=0x5555566b6848) at /localdisk/pakurapo/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/9.5.0/bits/hashtable.h:1567
#1  std::_Hashtable<int, std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > >, std::allocator<std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > >, std::__detail::_Select1st, std::equal_to<int>, std::hash<int>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_find_node (__c=0, __key=@0x55555709c378: 0, __bkt=0, this=0x5555566b6848)
    at /localdisk/pakurapo/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/9.5.0/bits/hashtable.h:687
#2  std::_Hashtable<int, std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > >, std::allocator<std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > >, std::__detail::_Select1st, std::equal_to<int>, std::hash<int>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_emplace<int&, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > (this=this@entry=0x5555566b6848) at /localdisk/pakurapo/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/9.5.0/bits/hashtable.h:1687
#3  0x00007fffa25f55a9 in std::_Hashtable<int, std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > >, std::allocator<std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > >, std::__detail::_Select1st, std::equal_to<int>, std::hash<int>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::emplace<int&, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > (this=0x5555566b6848) at /localdisk/pakurapo/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/9.5.0/bits/hashtable.h:780
#4  std::unordered_map<int, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> >, std::hash<int>, std::equal_to<int>, std::allocator<std::pair<int const, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > > >::emplace<int&, std::unique_ptr<ArrowStorage::TableData, std::default_delete<ArrowStorage::TableData> > > (this=0x5555566b6848) at /localdisk/pakurapo/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/9.5.0/bits/unordered_map.h:389
#5  ArrowStorage::createTable (this=0x5555566b6700, table_name=..., columns=..., options=...) at /localdisk/pakurapo/hdk/omniscidb/ArrowStorage/ArrowStorage.cpp:398
#6  0x00007fffa26740e0 in __pyx_pf_5pyhdk_8_storage_12ArrowStorage_2createTable (__pyx_v_self=0x7fffa12065d0, __pyx_v_table_name=0x7ffff69033f0, __pyx_v_scheme=<optimized out>, __pyx_v_table_opts=0x7fffb0d18670) at /localdisk/pakurapo/hdk/python/pyhdk/_storage.cpp:9737
#7  0x00007fffa267511f in __pyx_pw_5pyhdk_8_storage_12ArrowStorage_3createTable (__pyx_v_self=0x7fffa12065d0, __pyx_args=0x7fffb0d23f00, __pyx_kwds=<optimized out>) at /localdisk/pakurapo/hdk/python/pyhdk/_storage.cpp:9249
```

The failure happens when accessing the map of tables here https://github.com/intel-ai/hdk/blob/main/omniscidb/ArrowStorage/ArrowStorage.cpp#L398. Its size equals 72057598332895232 (which looks like `(schema_id << 24) + 1` and 4 zeroed bytes) when we enter the `createTable` call. It changes if I reorder class members. So the problem is somewhere between the constructor (it has size==0) and the `createTable` call.

**Comment by kurapov-peter** [comment]
Tried to use an asan-enabled HDK build with Folly. Folly's initialization fails with a segfault:
```
#0  0x00007fff65f08848 in std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_realloc_insert<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (this=0x7fffffff2320, __position=...) at /home/pakurapo/miniconda3/envs/omnisci-dev/x86_64-conda-linux-gnu/include/c++/12.2.0/bits/vector.tcc:439
#1  0x00007fff5da225d6 in std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::emplace_back<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (this=0x7fffffff2320) at /home/conda/feedstock_root/build_artifacts/folly_1667818832997/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/vector.tcc:121
#2  folly::CacheLocality::readFromProcCpuinfo () at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:285
#3  0x00007fff5da22687 in folly::getSystemLocalityInfo () at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:42
#4  folly::CacheLocality::system<std::atomic> () at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:39
#5  0x00007fff5da1fd89 in folly::detail::AccessSpreaderBase::initialize (state=..., pickGetcpuFunc=@0x7fff5d9b4380: {folly::Getcpu::Func (void)} 0x7fff5d9b4380 <folly::AccessSpreader<std::atomic>::pickGetcpuFunc()>, 
    system=@0x7fff5da22630: {const folly::CacheLocality &(void)} 0x7fff5da22630 <folly::CacheLocality::system<std::atomic>()>) at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:368
#6  0x00007fff5d9945c9 in folly::AccessSpreader<std::atomic>::initialize (state=...) at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.h:361
#7  folly::AccessSpreader<std::atomic>::state () at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.h:266
#8  folly::AccessSpreader<std::atomic>::state () at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.h:263
#9  folly::detail::AccessSpreaderStaticInit::AccessSpreaderStaticInit (this=0x7fff5dd24f80 <folly::detail::AccessSpreaderStaticInit::instance>) at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:358
#10 __static_initialization_and_destruction_0 (__priority=65535, __initialize_p=1) at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:360
#11 _GLOBAL__sub_I_CacheLocality.cpp(void) () at /usr/local/src/conda/folly-2022.11.07.00/folly/concurrency/CacheLocality.cpp:434
#12 0x00007ffff7fc947e in call_init (l=<optimized out>, argc=argc@entry=5, argv=argv@entry=0x7fffffffc588, env=env@entry=0x55555594d1e0) at ./elf/dl-init.c:70
```

**Pull request closed without merging**